core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -1264,6 +1270,7 @@ class DAGScheduler(
             try { // cancelTasks will fail if a SchedulerBackend does not implement killTask
               taskScheduler.cancelTasks(stageId, shouldInterruptThread)
               stage.latestInfo.stageFailed(failureReason)
+              outputCommitCoordinator.stageEnd(stage.id)
```

Comment:
Wouldn't it be better to put these three lines (at least) in a separate method:

    private def endStage(stage: Stage): Unit = {
      stage.latestInfo.stageFailed(failureReason)
      outputCommitCoordinator.stageEnd(stage.id)
      listenerBus.post(SparkListenerStageCompleted(stage.latestInfo))
    }

That would make it harder to miss things like this.